### PR TITLE
Added X-Siteaccess specification for REST doc

### PR DESCRIPTION
**Pull request by mateuszbieniek**
As in the tile.

### CODE REVIEW

**Comment by andrerom** [comment]
_Needs approval from doc team, other then that ready for merge._

File: doc/specifications/rest/REST-API-V2.rst [hunk 1]
```diff
@@ -9,6 +9,22 @@ eZ Platform REST API V2
 General considerations
 ======================
 
+Specifying a siteaccess
+-----------
+
+In order to specify a siteaccess when talking to the REST API, a custom header, "X-Siteaccess", needs to be provided.
```
Comment (michalmilc90) [inline doc/specifications/rest/REST-API-V2.rst]:
```suggestion
In order to specify a SiteAccess when talking to the REST API, a custom `X-Siteaccess` header has to be provided.
```

Comment (andrerom) [reply doc/specifications/rest/REST-API-V2.rst]: @mateuszbieniek seems you need to apply this :)

@michalmilc90 Does this mean this is approved from Doc team as well once this is in? ;)

Comment (michalmilc90) [reply doc/specifications/rest/REST-API-V2.rst]: @andrerom yes! 😉

File: doc/specifications/rest/REST-API-V2.rst [hunk 2]
```diff
@@ -9,6 +9,22 @@ eZ Platform REST API V2
 General considerations
 ======================
 
+Specifying a siteaccess
```
Comment (michalmilc90) [inline doc/specifications/rest/REST-API-V2.rst]:
```suggestion
Specifying a SiteAccess
```

**Comment by alongosz** [comment]
@ezsystems/documentation-team we need this also in upcoming RAML doc for REST.

**Pull request merged**